src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -146,11 +146,17 @@ const EditShiftHours = ({
     );
 
     const endTime = formData[`${inputName}-finish-time`] || null;
+    const endDate = `${formData['finishDate-year']}-${formData['finishDate-month']}-${formData['finishDate-day']}`;
     let actualEndDateTime = '';
     if (endTime) {
-      const actualEndDate = timeEntry.finishNextDay
-        ? getFinishTimeDate(actualStartDate)
-        : formatDate(localEndDate);
+      let actualEndDate;
+      if (isChecked) {
+        actualEndDate = formatDate(endDate);
```

Comment:
This will only save the new end date while the checkboxes are open - just checking that's the intended functionality?